### ISSUE TITLE
Blade directives mixed in with script tags causes formatter to fail

### ISSUE THREAD

**Issue by yasinallana1** [shufo/vscode-blade-formatter#47]
## Template

```
@push('scripts')
    <script src="{{ asset('backend/plugins/dropzone/dropzone.min.js') }}"></script>
    <script>
        Dropzone.options.gallery = {
            url: "...",
            previewTemplate: $("#preview-template").html(),
            dictDefaultMessage: "Drop files here or click to select",
            addRemoveLinks: true,
            uploadMultiple: true,
            maxFilesize: 2,
            filesizeBase: 1024,
            acceptedFiles: "image/jpg,image/jpeg,image/png",
            init: function () {
                this.renderExistingServerFiles = function(files, fileUrls , response){
                    for (const file in files) {
                        if (Object.hasOwnProperty.call(files, file)) {
                            const element = files[file];
                            
                            this.files.push(element);
                            this.displayExistingFile(element, fileUrls[file], null, null, true);
                            this.emit("processing", element);
                            this.emit("complete", element);
                        }
                    }
                    this.emit("successmultiple", files, response, false);
                }

                @isset($item->images)
                    let files = [];
                    let fileUrls = [];
                    let response = { status:"success", fileHashes: [], imageIds: []};
                    @foreach($item->images as $image)
                        @php
                            $imageUrl = asset('img/'.$image->image_hash);
                            $imagePath = public_path('img/'.$image->image_hash);
                            $imageMime = File::mimeType($imagePath);
                            $imageSize = File::size($imagePath);
                            $imageName = File::name($imagePath);
                        @endphp

                        files.push(
                            {
                                processing: true,
                                accepted: true,
                                name: "{{ $imageName }}",
                                size: {{ $imageSize }},
                                type: '{{ $imageMime }}',
                                status: Dropzone.SUCCESS,
                            },
                        );
                        
                        fileUrls.push("{{ $imageUrl }}")
                        response.fileHashes.push("{{ $image->image_hash }}")
                        response.imageIds.push("{{ $image->id }}")
                    @endforeach
                    this.renderExistingServerFiles(files, fileUrls, response);
                @endisset
            },
        }

    </script>
@endpush
```

## Context (Environment)

- Standard laravel dev environment

## Tell me what should happen

The formatter should be able to format blade code inside script tags.

## Description

Blade directives used inside java script tags causes issues with the formatter thereby causing it to fail.

**Comment by sadmachine** [comment]
I would also like to see proper formatting of Blade directives within script tags, or at least an "ignore" feature within script tags so that it doesn't automatically format it as garbage (similar to something like prettiers ''). One huge use case of blade is passing data to javascript to create reactive layouts.

**Comment by shufo** [comment]
@yasinallana1 
Fixed this behaviour on [v0.7.8](https://github.com/shufo/vscode-blade-formatter/releases/tag/v0.7.8) ([v1.11.1](https://github.com/shufo/blade-formatter/releases/tag/v1.11.1) on blade-formatter)

```blade
@push('scripts')
    <script src="{{ asset('backend/plugins/dropzone/dropzone.min.js') }}"></script>
    <script>
        Dropzone.options.gallery = {
            url: "...",
            previewTemplate: $("#preview-template").html(),
            dictDefaultMessage: "Drop files here or click to select",
            addRemoveLinks: true,
            uploadMultiple: true,
            maxFilesize: 2,
            filesizeBase: 1024,
            acceptedFiles: "image/jpg,image/jpeg,image/png",
            init: function() {
                this.renderExistingServerFiles = function(files, fileUrls, response) {
                    for (const file in files) {
                        if (Object.hasOwnProperty.call(files, file)) {
                            const element = files[file];

                            this.files.push(element);
                            this.displayExistingFile(element, fileUrls[file], null, null, true);
                            this.emit("processing", element);
                            this.emit("complete", element);
                        }
                    }
                    this.emit("successmultiple", files, response, false);
                }

                @isset($item->images)
                    let files = [];
                    let fileUrls = [];
                    let response = { status:"success", fileHashes: [], imageIds: []};
                    @foreach ($item->images as $image)
                        @php
                            $imageUrl = asset('img/' . $image->image_hash);
                            $imagePath = public_path('img/' . $image->image_hash);
                            $imageMime = File::mimeType($imagePath);
                            $imageSize = File::size($imagePath);
                            $imageName = File::name($imagePath);
                        @endphp

                        var order = {{ @json($array['item']->getOrder()) }}

                        files.push({
                        processing: true,
                        accepted: true,
                        name: "{{ $imageName }}",
                        size: {{ $imageSize }},
                        type: '{{ $imageMime }}',
                        status: Dropzone.SUCCESS,});

                        fileUrls.push("{{ $imageUrl }}")
                        response.fileHashes.push("{{ $image->image_hash }}")
                        response.imageIds.push("{{ $image->id }}")
                    @endforeach
                    this.renderExistingServerFiles(files, fileUrls, response);
                @endisset
            },
        }

    </script>
@endpush
```

Now you can format directive and raw javascript mixed template.

@sadmachine 
>`var order = @json($array['item']->getOrder());`

is also ok.
